mmv1/third_party/terraform/fwprovider/data_source_provider_config_plugin_framework.go

```diff
@@ -202,23 +203,52 @@ func (d *GoogleProviderConfigPluginFrameworkDataSource) Read(ctx context.Context
 	}
 
 	// Copy all values from the provider config into this data source
+	data.Credentials = types.StringValue(d.providerConfig.Credentials)
+	data.AccessToken = types.StringValue(d.providerConfig.AccessToken)
+	data.ImpersonateServiceAccount = types.StringValue(d.providerConfig.ImpersonateServiceAccount)
 
-	data.Credentials = d.providerConfig.Credentials
-	data.AccessToken = d.providerConfig.AccessToken
-	data.ImpersonateServiceAccount = d.providerConfig.ImpersonateServiceAccount
-	data.ImpersonateServiceAccountDelegates = d.providerConfig.ImpersonateServiceAccountDelegates
-	data.Project = d.providerConfig.Project
-	data.Region = d.providerConfig.Region
-	data.BillingProject = d.providerConfig.BillingProject
-	data.Zone = d.providerConfig.Zone
-	data.UniverseDomain = d.providerConfig.UniverseDomain
-	data.Scopes = d.providerConfig.Scopes
-	data.UserProjectOverride = d.providerConfig.UserProjectOverride
-	data.RequestReason = d.providerConfig.RequestReason
-	data.RequestTimeout = d.providerConfig.RequestTimeout
-	data.DefaultLabels = d.providerConfig.DefaultLabels
-	data.AddTerraformAttributionLabel = d.providerConfig.AddTerraformAttributionLabel
-	data.TerraformAttributionLabelAdditionStrategy = d.providerConfig.TerraformAttributionLabelAdditionStrategy
+	delegateAttrs := make([]attr.Value, len(d.providerConfig.ImpersonateServiceAccountDelegates))
```

Comment:
to double check, this extended logic is from the need to apply the string type assertions to every item within a list (as well as the list type on the list itself)?

seperately I think it would be good to have a comment somewhere around here summarizing the need for these assertions and the resulting mix of SDK behavior in this data source (as was mentioned [here](https://github.com/GoogleCloudPlatform/magic-modules/pull/11903#discussion_r1829448872), albeit not user-facing

Reply:
Here's a preview of that: https://github.com/SarahFrench/magic-modules/pull/10

Reply:
Done in https://github.com/GoogleCloudPlatform/magic-modules/pull/11903/commits/2ef48bbd94b82ccd1bdb2b5da3ac9c5431fbbeaf